internal/manifest/ready_check.go

```diff
@@ -51,40 +52,128 @@ func (c *CustomResourceReadyCheck) Run(ctx context.Context,
-		strings.Split(customResourceStatePath, ".")...)
+	return HandleState(manifest, moduleCR)
+}
+
+func HandleState(manifest *v1beta2.Manifest, moduleCR *unstructured.Unstructured) (declarative.StateInfo, error) {
+	typedState, stateExists, err := mappingState(manifest, moduleCR)
 	if err != nil {
-		return declarative.StateInfo{State: declarative.StateError},
-			fmt.Errorf("could not get state from module CR %s at path %s to determine readiness: %w",
-				res.GetName(), customResourceStatePath, ErrNoDeterminedState,
-			)
+		// Only happens for kyma module CR
+		if errors.Is(err, ErrNotSupportedState) {
+			return declarative.StateInfo{
+				State: declarative.StateWarning,
+				Info:  ErrNotSupportedState.Error(),
+			}, nil
+		}
+		return declarative.StateInfo{State: declarative.StateError}, fmt.Errorf(
+			"could not get state from module CR %s to determine readiness: %w",
+			moduleCR.GetName(), err,
+		)
 	}
 
 	// CR state might not been initialized, put manifest state into processing
 	if !stateExists {
 		return declarative.StateInfo{State: declarative.StateProcessing, Info: "module CR state not found"}, nil
 	}
 
-	typedState := declarative.State(stateFromCR)
-
-	if !typedState.IsSupportedState() {
-		return declarative.StateInfo{
-			State: declarative.StateWarning,
-			Info:  "module CR state is not supported, this module might not be default kyma module",
-		}, nil
-	}
-
 	if typedState == declarative.StateDeleting || typedState == declarative.StateError {
 		return declarative.StateInfo{State: typedState}, ErrCRInUnexpectedState
 	}
 	return declarative.StateInfo{State: typedState}, nil
 }
 
+func mappingState(manifest *v1beta2.Manifest, moduleCR *unstructured.Unstructured) (declarative.State, bool, error) {
+	stateChecks, customStateFound, err := parseStateChecks(manifest)
+	if err != nil {
+		return "", false, err
+	}
+	// make sure ready and error state exists, for other missing customized state, can be ignored.
+	if requiredStateMissing(stateChecks) {
+		return "", false, ErrRequiredStateMissing
+	}
+	stateResult := map[v1beta2.State]bool{}
+	for _, stateCheck := range stateChecks {
+		stateFromCR, stateExists, err := unstructured.NestedString(moduleCR.Object,
+			strings.Split(stateCheck.JSONPath, ".")...)
+		if err != nil {
+			return "", false, fmt.Errorf("could not get state from module CR %s at path %s "+
+				"to determine readiness: %w", moduleCR.GetName(), stateCheck.JSONPath, err)
+		}
+		if !stateExists {
+			continue
+		}
+		if !customStateFound && !declarative.State(stateFromCR).IsSupportedState() {
+			return "", false, ErrNotSupportedState
+		}
+		_, found := stateResult[stateCheck.MappedState]
+		if found {
+			stateResult[stateCheck.MappedState] = stateResult[stateCheck.MappedState] && stateFromCR == stateCheck.Value
+		} else {
+			stateResult[stateCheck.MappedState] = stateFromCR == stateCheck.Value
+		}
+	}
+	return calculateFinalState(stateResult), true, nil
```

Comment:
Can you define the "jsonPaths are configured wrong" a bit more concretely? If you mean the jsonPath can't be found, then it will fail in the `stateFromCR, stateExists, err := unstructured.NestedString`.

Reply:
yes I mean it can't be found. I tried a non-existing path, but the err was still nil here

Reply:
ok, good point, that's valid, which means
```
if !stateExists {
   continue
}
```
should not continue, it's better we indicate user this provided path not exists.

Reply:
I decided to check against the manifest CR creation timestamp to know when to put processing or warning state, please check here https://github.com/kyma-project/lifecycle-manager/pull/784/commits/2928408e13715720b2cd4dc3c924e8f4661736ac#diff-98067b6967c5fe86151bd1d8629e4107ff72feef3ac1a237c4b56e6eb2ad14e0R95-R99